### PR TITLE
Token Revocation on different events

### PR DESCRIPTION
**Pull request by sadilchamishka**
### Proposed changes in this pull request

$subject

### CODE REVIEW

**Comment by sadilchamishka** [comment]
The below test cases were failed to a mistake of the initial commit.

Screenshot 2024-02-27 at 21 46 25

Verified that failed test cases are passing with the fix by this commit - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2382/commits/299d8569bc66d62bf6d1aa8092e7a968f95c0e91.

Screenshot 2024-02-27 at 21 44 31

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java [hunk 1]
```diff
@@ -693,36 +701,91 @@ public static boolean revokeAuthzCodes(String username, UserStoreManager userSto
...
+        try {
+            String accessingOrg = OAuthComponentServiceHolder.getInstance().getOrganizationManager()
+                    .resolveOrganizationId(tenantDomain);
+            if (managedOrg != null) {
+                tenantDomain = OAuthComponentServiceHolder.getInstance().getOrganizationManager()
+                        .resolveTenantDomain(managedOrg);
+                authenticatedUser.setUserResidentOrganization(managedOrg);
+            } else {
+                authenticatedUser.setFederatedUser(true);
+                authenticatedUser.setFederatedIdPName("SSO");
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java:742]: This should be resolved. Usually this won't changed by users

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/SharedAppResolveDAO.java [hunk 2]
```diff
@@ -54,4 +56,22 @@ public static String resolveSharedApplication(String appResideOrgId, String main
                     ERROR_CODE_ERROR_RESOLVING_SHARED_APPLICATION.getMessage(), e);
         }
     }
+
+    public static String getMainApplication(String sharedAppId, String sharedOrgId) throws IdentityOAuth2Exception {
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/SharedAppResolveDAO.java:60]: move this code to org-mgt-core as services

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java [hunk 3]
```diff
@@ -693,36 +704,98 @@ public static boolean revokeAuthzCodes(String username, UserStoreManager userSto
...
+            if (authenticatedUser.getUserResidentOrganization() != null) {
+                List<String> newAssociatedApplications = new ArrayList<>();
+                for (String app : associatedApplications) {
+                    newAssociatedApplications.add(
+                            SharedAppResolveDAO.getMainApplication(app, authenticatedUser.getAccessingOrganization()));
+                }
+                associatedApplications = newAssociatedApplications;
+            }
+        } catch (IdentityOAuth2Exception e) {
+            throw new RuntimeException(e);
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java:792]: Fix this

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java [hunk 4]
```diff
@@ -496,7 +498,12 @@ public AccessTokenDO getLatestAccessToken(String consumerKey, AuthenticatedUser
 
             prepStmt = connection.prepareStatement(sql);
             prepStmt.setString(1, getPersistenceProcessor().getProcessedClientId(consumerKey));
+            String rootTenantDomain = (String) IdentityUtil.threadLocalProperties.get()
+                    .get(OrganizationManagementConstants.ROOT_TENANT_DOMAIN);
             int appTenantId = IdentityTenantUtil.getLoginTenantId();
+            if (rootTenantDomain != null) {
```
Comment (sadilchamishka) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/AccessTokenDAOImpl.java:504]: check revocation of sub-org token. is it working.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java (hunk 3 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java]: Fixed by - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2382/commits/401d9c67e6831c1ae56ccafb4d004920242e713c

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java [hunk 5]
```diff
@@ -864,6 +939,17 @@ private static boolean processTokenRevocation(Set<String> clientIds, Authenticat
...
     }
 
+    private static String resolveRootTenantDomain(String tenantDomain) {
+
+        try {
+            String rootOrgId = OAuthComponentServiceHolder.getInstance().getOrganizationManager()
+                    .getPrimaryOrganizationId(tenantDomain);
+            return OAuthComponentServiceHolder.getInstance().getOrganizationManager().resolveTenantDomain(rootOrgId);
+        } catch (OrganizationManagementException e) {
+            throw new RuntimeException(e);
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java:949]: make this a checked exception

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java [hunk 6]
```diff
@@ -693,36 +704,100 @@ public static boolean revokeAuthzCodes(String username, UserStoreManager userSto
...
+            if (authenticatedUser.getUserResidentOrganization() != null) {
+                List<String> newAssociatedApplications = new ArrayList<>();
+                for (String app : associatedApplications) {
+                    newAssociatedApplications.add(
+                            SharedAppResolveDAO.getMainApplication(app, authenticatedUser.getAccessingOrganization()));
+                }
+                associatedApplications = newAssociatedApplications;
+            }
+        } catch (IdentityOAuth2Exception e) {
+            LOG.error("Error occurred while getting the main applications of the shared apps.", e);
```
Comment (janakamarasena) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java:793]: remove the log

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java [hunk 7]
```diff
@@ -5236,4 +5237,20 @@ public static String getMyAccountCallbackFromServerConfig(String tenantDomain) {
         }
         return null;
     }
+
+    public static String getUserResidentTenantDomain(AuthenticatedUser authenticatedUser)
```
[the remark this answers is not in the document]
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java:5241]: Fixed by -https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2382/commits/3a73357862da03385d0d13f543d385b492fb314d

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java (hunk 6 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java]: Fixed by - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2382/commits/3a73357862da03385d0d13f543d385b492fb314d

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java (hunk 5 above)
Comment (sadilchamishka) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthUtil.java]: Fixed by - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2382/commits/3a73357862da03385d0d13f543d385b492fb314d

**Review by janakamarasena — APPROVED** [review]

**Pull request merged**